### PR TITLE
Homework 3 Pipe

### PR DESCRIPTION
**Pull request by Maagikus**
Was created pipe for transforming film duration to hh:mm:ss format, the interface movie has been epanded for this task (duration field).
A linter was added to the project and all files were formatted according to the rules.

### CODE REVIEW

File: src/app/components/movie-list/movie-list.component.html [hunk 1]
```diff
@@ -1,33 +1,37 @@
...
-				<div class="movie-list__message">Ви не обрали фільмів для перегляду</div>
-			</ng-template>
-		</div>
-	</div>
-</div>
\ No newline at end of file
+  <div class="movie-list__content">
+    <div class="movie-list__all">
+      <ul class="movie-list__list">
+        <li class="movie-list__item" *ngFor="let movie of movieList">
```
Comment (inga-lekhman) [inline src/app/components/movie-list/movie-list.component.html]: TrackBy забули додати)

File: src/app/components/movie-list/movie-list.component.html [hunk 2]
```diff
@@ -1,33 +1,37 @@
...
+            (addFavorite)="onAddToFavorites($event)"
+            (addWatchList)="onAddToWatchList($event)"
+          ></app-movie-card>
+        </li>
+      </ul>
+    </div>
+    <div class="movie-list__favourite">
+      <div class="movie-list__title">Улюблені</div>
+      <ul *ngIf="favoriteList.length; else noMovies" class="movie-list__list">
+        <li class="movie-list__item-normal" *ngFor="let movie of favoriteList">
```
Comment (inga-lekhman) [inline src/app/components/movie-list/movie-list.component.html]: додайте TrackBy

File: src/app/components/movie-list/movie-list.component.html [hunk 3]
```diff
@@ -1,33 +1,37 @@
...
+        </li>
+      </ul>
+      <ng-template #noMovies>
+        <div class="movie-list__message">Ви не обрали улюблених фільмів</div>
+      </ng-template>
+    </div>
+    <div class="movie-list__watch">
+      <div class="movie-list__title">До перегляду</div>
+      <ul *ngIf="watchList.length; else noMovies" class="movie-list__list">
+        <li class="movie-list__item-normal" *ngFor="let movie of watchList">
```
Comment (inga-lekhman) [inline src/app/components/movie-list/movie-list.component.html]: додайте TrackBy

File: src/app/pipes/text-splitter/text-truncate.pipe.ts [hunk 4]
```diff
@@ -10,6 +10,7 @@ export class TextTruncatePipe implements PipeTransform {
     if (value.length < MAX_LENGTH) return value;
     const truncatedString = value.substring(0, MAX_LENGTH);
     const lastSpaceIndex = truncatedString.lastIndexOf(' ');
+
     return lastSpaceIndex === -1
       ? truncatedString
       : truncatedString.substring(0, lastSpaceIndex) + '...';
```
Comment (inga-lekhman) [inline src/app/pipes/text-splitter/text-truncate.pipe.ts:16]: Гарна ідея для pipe!

File: src/app/components/movie-list/movie-list.component.html (hunk 1 above)
Comment (Maagikus) [reply src/app/components/movie-list/movie-list.component.html]: Додав

File: src/app/components/movie-list/movie-list.component.html (hunk 2 above)
Comment (Maagikus) [reply src/app/components/movie-list/movie-list.component.html]: Додав

File: src/app/components/movie-list/movie-list.component.html (hunk 3 above)
Comment (Maagikus) [reply src/app/components/movie-list/movie-list.component.html]: Додав

**Pull request merged**